drivers/ethernet/eth_e1000.c

```diff
@@ -207,20 +207,16 @@ static void e1000_isr(const struct device *device)
 	}
 }
 
-#define PCI_VENDOR_ID_INTEL	0x8086
-#define PCI_DEVICE_ID_I82540EM	0x100e
-
 DEVICE_DECLARE(eth_e1000);
 
 int e1000_probe(const struct device *device)
 {
-	const pcie_bdf_t bdf = PCIE_BDF(0, 3, 0);
+	const pcie_bdf_t bdf = DT_INST_REG_ADDR(0);
 	struct e1000_dev *dev = device->data;
 	uint32_t ral, rah;
 	struct pcie_mbar mbar;
 
-	if (!pcie_probe(bdf, PCIE_ID(PCI_VENDOR_ID_INTEL,
-				     PCI_DEVICE_ID_I82540EM))) {
+	if (!pcie_probe(bdf, DT_INST_REG_SIZE(0))) {
```

Comment:
It seems like leaving the PCI_VENDOR_ID_INTEL/PCI_DEVICE_ID_I82540EM hardcoded to the driver makes sense, and not getting that from DTS.

Reply:
I do not understand the logic behind this comment.